clients/client/src/main/java/org/projectnessie/client/ResultStreamPaginator.java

```diff
@@ -54,6 +54,16 @@
     this.fetcher = fetcher;
   }
 
+  Stream<E> limitedStream(OptionalInt maxRecords) throws NessieNotFoundException {
+    return limitedStream(maxRecords, OptionalInt.empty());
+  }
+
+  Stream<E> limitedStream(OptionalInt maxRecords, OptionalInt pageSizeHint)
+      throws NessieNotFoundException {
+    Stream<E> stream = generateStream(pageSizeHint);
```

Comment:
Not a concern of this PR, because the added `.stream()` functions don't allow setting the page-size-hint.
Mind providing a test for this, since you raised the issue?

Reply:
I feel it is a concern as we are exposing a buggy interface to the user now.

Testcase is already there 
https://github.com/projectnessie/nessie/blob/main/servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractRestCommitLog.java#L401

just need to add an assert that `completeLog.size() == pageSizeHint` to see the issue.